### ISSUE TITLE
defineModuleAsClassTest_0 - ClassFormatError: JVMCFRE152 module-info is not a class because access_flag ACC_MODULE is set; class=module-info, offset=27

### ISSUE THREAD

**Issue by JasonFengJ9** [eclipse-openj9/openj9#12629]
Failure link
------------

From an internal build `Test_openjdk11_j9_extended.functional_ppc64le_linux_testList_0/36/`:
```
java version "11.0.11" 2021-04-20
Java(TM) SE Runtime Environment 11.0.10.0-IBM (build 11.0.11+9-202105050106)
Eclipse OpenJ9 VM 11.0.10.0-IBM (build master-764252c83, JRE 11 Linux ppc64le-64-Bit Compressed References 20210504_80 (JIT enabled, AOT enabled)
OpenJ9   - 764252c83
OMR      - 79c52c726
JCL      - da466bb02c based on jdk-11.0.11+9)
```
[Rerun in Grinder](https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/parambuild/?TARGET=defineModuleAsClassTest_0&amp;SDK_RESOURCE=upstream&amp;TEST_FLAG=&amp;UPSTREAM_TEST_JOB_NAME=Test_openjdk11_j9_extended.functional_ppc64le_linux&amp;DOCKER_REQUIRED=false&amp;ACTIVE_NODE_TIMEOUT=0&amp;EXTRA_DOCKER_ARGS=&amp;VENDOR_TEST_DIRS=functional&amp;TKG_OWNER_BRANCH=AdoptOpenJDK%3Amaster&amp;PLATFORM=ppc64le_linux&amp;KEEP_REPORTDIR=false&amp;PERSONAL_BUILD=false&amp;ADOPTOPENJDK_REPO=https%3A%2F%2Fgithub.com%2FAdoptOpenJDK%2Fopenjdk-tests.git&amp;LABEL=&amp;EXTRA_OPTIONS=&amp;CUSTOMIZED_SDK_URL=+https%3A%2F%2Fna.artifactory.swg-devops.com%2Fartifactory%2Fsys-rt-generic-local%2Fhyc-runtimes-jenkins.swg-devops.com%2Fbuild-scripts%2Fjobs%2Fjdk11u%2Fjdk11u-linux-ppc64le-openj9-IBM%2F80%2FOpenJDK11U-testimage_ppc64le_linux_openj9_IBM_2021-05-05-01-01.tar.gz+https%3A%2F%2Fna.artifactory.swg-devops.com%2Fartifactory%2Fsys-rt-generic-local%2Fhyc-runtimes-jenkins.swg-devops.com%2Fbuild-scripts%2Fjobs%2Fjdk11u%2Fjdk11u-linux-ppc64le-openj9-IBM%2F80%2FOpenJDK11U-debugimage_ppc64le_linux_openj9_IBM_2021-05-05-01-01.tar.gz+https%3A%2F%2Fna.artifactory.swg-devops.com%2Fartifactory%2Fsys-rt-generic-local%2Fhyc-runtimes-jenkins.swg-devops.com%2Fbuild-scripts%2Fjobs%2Fjdk11u%2Fjdk11u-linux-ppc64le-openj9-IBM%2F80%2FOpenJDK11U-jdk_ppc64le_linux_openj9_IBM_2021-05-05-01-01.tar.gz+https%3A%2F%2Fna.artifactory.swg-devops.com%2Fartifactory%2Fsys-rt-generic-local%2Fhyc-runtimes-jenkins.swg-devops.com%2Fbuild-scripts%2Fjobs%2Fjdk11u%2Fjdk11u-linux-ppc64le-openj9-IBM%2F80%2FOpenJDK11U-jre_ppc64le_linux_openj9_IBM_2021-05-05-01-01.tar.gz&amp;BUILD_IDENTIFIER=&amp;ADOPTOPENJDK_BRANCH=master&amp;LIGHT_WEIGHT_CHECKOUT=true&amp;ARTIFACTORY_SERVER=na.artifactory.swg-devops&amp;KEEP_WORKSPACE=false&amp;JDK_VERSION=11&amp;USER_CREDENTIALS_ID=83181e25-eea4-4f55-8b3e-e79615733226&amp;ITERATIONS=1&amp;VENDOR_TEST_REPOS=git%40github.ibm.com%3Aruntimes%2Ftest.git&amp;JDK_REPO=git%40github.com%3Aibmruntimes%2Fopenj9-openjdk-jdk11&amp;RELEASE_TAG=&amp;OPENJ9_BRANCH=master&amp;OPENJ9_SHA=&amp;JCK_GIT_REPO=&amp;VENDOR_TEST_BRANCHES=master&amp;OPENJ9_REPO=https%3A%2F%2Fgithub.com%2Feclipse%2Fopenj9.git&amp;UPSTREAM_JOB_NAME=&amp;CUSTOM_TARGET=&amp;VENDOR_TEST_SHAS=&amp;JDK_BRANCH=openj9&amp;LABEL_ADDITION=&amp;ARTIFACTORY_REPO=&amp;ARTIFACTORY_ROOT_DIR=&amp;UPSTREAM_TEST_JOB_NUMBER=61&amp;DOCKERIMAGE_TAG=&amp;JDK_IMPL=openj9&amp;SSH_AGENT_CREDENTIAL=83181e25-eea4-4f55-8b3e-e79615733226&amp;AUTO_DETECT=true&amp;TKG_SHA=&amp;CUSTOMIZED_SDK_URL_CREDENTIAL_ID=4e18ffe7-b1b1-4272-9979-99769b68bcc2&amp;NUM_MACHINES=&amp;OPENJDK_SHA=&amp;BUILD_LIST=functional&amp;UPSTREAM_JOB_NUMBER=&amp;TIME_LIMIT=10&amp;JVM_OPTIONS=&amp;PARALLEL=None)

Optional info
-------------

Failure output (captured from console output)
---------------------------------------------
```
[2021-05-05T03:04:25.678Z] Running test defineModuleAsClassTest_0 ...
[2021-05-05T03:04:25.678Z] ===============================================
[2021-05-05T03:04:25.678Z] defineModuleAsClassTest_0 Start Time: Tue May  4 20:04:24 2021 Epoch Time (ms): 1620183864531
[2021-05-05T03:04:25.678Z] "/home/jenkins/workspace/Test_openjdk11_j9_extended.functional_ppc64le_linux_testList_0/openjdkbinary/j2sdk-image/bin/java" -Xshareclasses:destroyAll; "/home/jenkins/workspace/Test_openjdk11_j9_extended.functional_ppc64le_linux_testList_0/openjdkbinary/j2sdk-image/bin/java" -Xshareclasses:groupAccess,destroyAll; echo "cache cleanup done";
[2021-05-05T03:04:25.678Z] JVMSHRC005I No shared class caches available
[2021-05-05T03:04:25.678Z] JVMSHRC005I No shared class caches available
[2021-05-05T03:04:25.678Z] cache cleanup done
[2021-05-05T03:04:25.678Z] variation: NoOptions
[2021-05-05T03:04:25.679Z] JVM_OPTIONS:  

[2021-05-05T03:04:27.894Z] FAILED: testLoadModuleInClassLoader
[2021-05-05T03:04:27.894Z] java.lang.ClassFormatError: JVMCFRE152 module-info is not a class because access_flag ACC_MODULE is set; class=module-info, offset=27
[2021-05-05T03:04:27.894Z] 	at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:440)
[2021-05-05T03:04:27.894Z] 	at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:392)
[2021-05-05T03:04:27.894Z] 	at org.openj9.test.defineModuleAsClass.MyClassLoader.loadClass(DefineModuleAsClassTest.java:74)
[2021-05-05T03:04:27.894Z] 	at org.openj9.test.defineModuleAsClass.DefineModuleAsClassTest.testLoadModuleInClassLoader(DefineModuleAsClassTest.java:58)
[2021-05-05T03:04:27.894Z] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[2021-05-05T03:04:27.894Z] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[2021-05-05T03:04:27.894Z] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2021-05-05T03:04:27.894Z] 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
[2021-05-05T03:04:27.894Z] 	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
[2021-05-05T03:04:27.894Z] 	at org.testng.internal.Invoker.invokeMethod(Invoker.java:580)
[2021-05-05T03:04:27.894Z] 	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:716)
[2021-05-05T03:04:27.894Z] 	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:988)
[2021-05-05T03:04:27.894Z] 	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
[2021-05-05T03:04:27.894Z] 	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
[2021-05-05T03:04:27.894Z] 	at org.testng.TestRunner.privateRun(TestRunner.java:648)
[2021-05-05T03:04:27.894Z] 	at org.testng.TestRunner.run(TestRunner.java:505)
[2021-05-05T03:04:27.894Z] 	at org.testng.SuiteRunner.runTest(SuiteRunner.java:455)
[2021-05-05T03:04:27.894Z] 	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:450)
[2021-05-05T03:04:27.894Z] 	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:415)
[2021-05-05T03:04:27.894Z] 	at org.testng.SuiteRunner.run(SuiteRunner.java:364)
[2021-05-05T03:04:27.894Z] 	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
[2021-05-05T03:04:27.894Z] 	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:84)
[2021-05-05T03:04:27.894Z] 	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1208)
[2021-05-05T03:04:27.894Z] 	at org.testng.TestNG.runSuitesLocally(TestNG.java:1137)
[2021-05-05T03:04:27.894Z] 	at org.testng.TestNG.runSuites(TestNG.java:1049)
[2021-05-05T03:04:27.894Z] 	at org.testng.TestNG.run(TestNG.java:1017)
[2021-05-05T03:04:27.894Z] 	at org.testng.TestNG.privateMain(TestNG.java:1354)
[2021-05-05T03:04:27.894Z] 	at org.testng.TestNG.main(TestNG.java:1323)
[2021-05-05T03:04:27.894Z] 
[2021-05-05T03:04:27.894Z] 
[2021-05-05T03:04:27.894Z] ===============================================
[2021-05-05T03:04:27.895Z]     defineModuleAsClassTest
[2021-05-05T03:04:27.895Z]     Tests run: 1, Failures: 1, Skips: 0
[2021-05-05T03:04:27.895Z] ===============================================
[2021-05-05T03:04:27.895Z] 
[2021-05-05T03:04:27.895Z] 
[2021-05-05T03:04:27.895Z] ===============================================
[2021-05-05T03:04:27.895Z] Java9andUp suite
[2021-05-05T03:04:27.895Z] Total tests run: 1, Failures: 1, Skips: 0
[2021-05-05T03:04:27.895Z] ===============================================
[2021-05-05T03:04:27.895Z] 
[2021-05-05T03:04:27.895Z] 
[2021-05-05T03:04:27.895Z] defineModuleAsClassTest_0_FAILED
```

This seems caused by https://github.com/eclipse-openj9/openj9/pull/12620
fyi @ChengJin01 @pshipton

**Comment by ChengJin01** [comment]
Hi @JasonFengJ9 ,  the code & test changes were merged to https://github.com/eclipse-openj9/openj9 rather than https://github.com/eclipse/openj9.git but the test grinder still stick to `eclipse/openj9.git` with the unchanged tests by the latest build with the code changes.

Please try a new grinder with https://github.com/eclipse-openj9/openj9.

**Comment by JasonFengJ9** [comment]
> but the test grinder still sticks to eclipse/openj9.git with the unchanged tests by the latest build with the code changes.

fyi @llxia

**Comment by llxia** [comment]
https://github.com/eclipse/openj9 should be redirected to https://github.com/eclipse-openj9/openj9 automatically. It should not be a problem to continue eclipse/openj9. 

From SHA.txt, we are running `8a6e1df5c0cc2200c3f07efc04cc62946c9cdac1`. It should contain the change [Update the error action for ACC_MODULE](https://github.com/eclipse-openj9/openj9/commit/b5f5102607cfe65db18f3878b430e2ff3a6a6ffb)
```
timestamp: 20210504-185720
repo dir: /home/jenkins/workspace/Test_openjdk11_j9_extended.functional_ppc64le_linux_testList_0/openjdk-tests/openj9
git repo: 
  Fetch URL: https://github.com/eclipse/openj9.git
sha:
8a6e1df5c0cc2200c3f07efc04cc62946c9cdac1
```

**Comment by ChengJin01** [comment]
If so, then the nightly build at didn't pick up the latest changes according to the output (it was supposed to throw out `NoClassDefFoundError` rather than `ClassFormatError`)

FYI @AdamBrousseau & @jdekonin

**Comment by llxia** [comment]
re https://github.com/eclipse-openj9/openj9/issues/12629#issuecomment-832822808, just to complete the testing story, Grinder/15284 used https://github.com/eclipse-openj9/openj9 and had the same failure.

**Comment by llxia** [comment]
It looks like the SDK build picked up https://github.com/eclipse-openj9/openj9/commit/764252c839f441bcc654491efe78dfb2fa76a92b and test build picked up https://github.com/eclipse-openj9/openj9/commit/8a6e1df5c0cc2200c3f07efc04cc62946c9cdac1. Giving the fact that there was only 1 hour in between the two commits. I suspect the commit 8a6e1df was merged in between the SDK build and the test build. 

In summary, tonight's build should not have this problem. The proper fix should be set openj9 SHA to test build, so SDK build and test build will not be out of sync.

**Comment by AdamBrousseau** [comment]
Note that this is from a build using Adopt's scripts. OpenJ9 build scripts sync the test material with the compile.
Opened https://github.com/adoptium/ci-jenkins-pipelines/issues/143 to address.

**Comment by pshipton** [comment]
Since we have the adoptiium issue to address the problem, closing this one.
